### PR TITLE
remove logrus logger and add zerolog Logger to flowkit backend config

### PR DESCRIPTION
**Pull request by bjartek**
in commit #a03e87a03 the emulator was updated without adding a way for thirdparties to set the logger. 

This pr removes logrus and adds zerolog to the EmulatorGateway so that it is possible again.

### CODE REVIEW

**Comment by bjartek** [comment]
It also looks like WithEmulatorLogger is not used anymore. It is the logger sent into the backend that is used for both logging meters and logging from emulator.

File: pkg/flowkit/gateway/emulator.go [hunk 1]
```diff
@@ -57,21 +56,21 @@ func NewEmulatorGatewayWithOpts(serviceAccount *flowkit.Account, opts ...func(*E
...
+		logger:          &zerolog.Logger{},
 		emulatorOptions: []emulator.Option{},
 	}
 	for _, opt := range opts {
 		opt(gateway)
 	}
 
 	gateway.emulator = newEmulator(serviceAccount, gateway.emulatorOptions...)
-	gateway.backend = backend.New(&zerolog.Logger{}, gateway.emulator)
+	gateway.backend = backend.New(gateway.logger, gateway.emulator)
```
Comment (janezpodhostnik) [inline pkg/flowkit/gateway/emulator.go:67]: Unrelated: 
  Hm... Interesting that zerolog is passed by reference. I'm not sure it should be. Something to look into...

File: pkg/flowkit/gateway/emulator.go [hunk 2]
```diff
@@ -57,21 +56,21 @@ func NewEmulatorGatewayWithOpts(serviceAccount *flowkit.Account, opts ...func(*E
 
 	gateway := &EmulatorGateway{
 		ctx:             context.Background(),
-		logger:          logrus.New(),
+		logger:          &zerolog.Logger{},
```
Comment (janezpodhostnik) [inline pkg/flowkit/gateway/emulator.go:59]:
```suggestion
		logger:          &zerolog.Nop(),
```

File: pkg/flowkit/gateway/emulator.go (hunk 1 above)
Comment (bluesign) [reply pkg/flowkit/gateway/emulator.go:67]: hmm I was thinking on this when moving from logrus to zerolog, but I felt reference passing is correct. But I maybe totally missing the point. Why do you think copy is better @janezpodhostnik  ?

Comment (sideninja) [reply pkg/flowkit/gateway/emulator.go:67]: I think the issue might be that the receiver can have an influence on the logging of the other components.

Comment (sideninja) [reply pkg/flowkit/gateway/emulator.go:67]: In any case I don't think this is the subject of this PR

**Review by sideninja — APPROVED** [review]

File: pkg/flowkit/gateway/emulator.go (hunk 1 above)
Comment (bluesign) [reply pkg/flowkit/gateway/emulator.go:67]: I think in that case pass by reference is still correct. Sender always have option to send a new logger, but on the other case, sender loses influence on the component. ( e.g. change logging level in runtime )

**Review by janezpodhostnik — APPROVED** [review]

**Comment by bjartek** [comment]
dammit what did I do to this pr now, i just merged in master, is this ok or should i clean it up?

**Comment by bjartek** [comment]
i tried to update master but it just created a mess so I will leave it like it is.

**Comment by sideninja** [comment]
@bjartek could you update changes from master so we can get this in.

**Comment by bjartek** [comment]
@sideninja master merged. feel free to squash into main

**Comment by sideninja** [comment]
@bjartek the tests fail, can you check what is the issue please.

**Pull request merged**